network.go

```diff
@@ -61,7 +61,7 @@ type NetworkController interface {
 	Networks() []Network
 
 	// WalkNetworks uses the provided function to walk the Network(s) managed by this controller.
-	WalkNetworks(walker NetworkWalker)
+	WalkNetworks(walker NetworkWalker) bool
```

Comment:
> no constraints are put on the intent or implementation of the walking function.

I don't understand how this adds any constraints on the intent or implementation of the walking function

This changes makes it easier for clients who want to use walk to find a network, and adds no complexity to other consumers of the API (they can just ignore the return value).  See this code:

https://github.com/tomwilkie/docker/blob/network_extensions_dev/daemon/network_lib.go#L49

I don't understand why you wouldn't want to make users of your code's life easier?